main/src/ca/uwaterloo/flix/language/ast/Scheme.scala

```diff
@@ -156,17 +156,19 @@ object Scheme {
     // General Case: Compute if `sc1` <= `sc2`.
     //
 
-    // Instantiate every variable in `sc1` as flexible and make every free variable rigid.
-    val (tconstrs1, tpe1) = instantiate(sc1, InstantiateMode.Mixed)
+    // Mark every free variable in `sc1` as rigid.
+    val renv1 = sc1.base.typeVars.map(_.sym) -- sc1.quantifiers
 
-    // Instantiate every variable in `sc2` as rigid and make every free variable rigid.
-    val (tconstrs2, tpe2) = instantiate(sc2, InstantiateMode.Rigid)
+    // Mark every free and bound variable in `sc2` as rigid.
+    val renv2 = sc2.base.typeVars.map(_.sym)
+
+    val renv = RigidityEnv(renv1 ++ renv2)
```

Comment:
Why not define ++ on RigidityEnv instead?

Reply:
I was thinking of `RigidityEnv` as conceptually a map tvar -> rigidity, rather than a set. Thinking this way, a concatenation operation doesn't match the semantics, since "conflicts" here are resolved by preferring rigid.

That said, it doesn't really matter. I'll add the operation.